### PR TITLE
multisend token function

### PR DESCRIPTION
**Pull request by rykci**
- admin mapping,
- functions to set and remove admin
- sendmultitoken function

### CODE REVIEW

File: smartcontract/SwanFaucet.sol [hunk 1]
```diff
@@ -37,4 +41,38 @@ contract SwanFaucet is OwnableUpgradeable {
...
+    }
+
+    // pass in array of token addresses and the amount array to send, with a receiving address
+    function sendMultiToken(address[] memory _tokenAddresses, uint[] memory tokenAmounts, address _address) public onlyAdmin {
+        require(allowedToWithdraw(msg.sender), "please wait 24 hours");
+        
+        nextAccessTime[_address] = block.timestamp + waitTime;
+
+        for(uint i=0; i<_tokenAddresses.length; i++) {
+            require(tokenAmounts[i] <= ERC20(_tokenAddresses[i]).balanceOf(address(this)));
```
Comment (liushmh) [inline smartcontract/SwanFaucet.sol]: can you change it to
```
if(tokenAmounts[i] <= ERC20(_tokenAddresses[i]).balanceOf(address(this)))
  ERC20(_tokenAddresses[i]).transfer(_address, tokenAmounts[i]);

```
Putting require(xx) here will stop transferring tokens which are enough to complete transfer

File: smartcontract/SwanFaucet.sol [hunk 2]
```diff
@@ -37,4 +41,38 @@ contract SwanFaucet is OwnableUpgradeable {
         }
         return false;
     }
+
+    function sendTokensTo(address _address) public {
```
Comment (liushmh) [inline smartcontract/SwanFaucet.sol]: add OnlyAdmin here

File: smartcontract/SwanFaucet.sol [hunk 3]
```diff
@@ -16,13 +18,15 @@ contract SwanFaucet is OwnableUpgradeable {
...
     mapping(address => uint256) nextAccessTime;
 
+    mapping(address => bool) isAdmin;
+
     function initialize(address _tokenInstance) public initializer {
         require(_tokenInstance != address(0));
         tokenInstance = ERC20(_tokenInstance);
     }
 
     function requestTokens() public {
```
Comment (liushmh) [inline smartcontract/SwanFaucet.sol:27]: I think we can remove this one.

File: smartcontract/SwanFaucet.sol [hunk 4]
```diff
@@ -6,6 +6,8 @@ import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 interface ERC20 {
     function transfer(address to, uint256 value) external returns (bool);
     event Transfer(address indexed from, address indexed to, uint256 value);
```
Comment (liushmh) [inline smartcontract/SwanFaucet.sol]: remove this event, since we don't generate event in this contract

File: smartcontract/SwanFaucet.sol (hunk 3 above)
Comment (rykci) [reply smartcontract/SwanFaucet.sol:27]: `mapping(address => bool) isAdmin;` i need this to keep track of admins

**Review by liushmh — APPROVED** [review]

**Pull request merged**